DeviceClassifier/OneLayer/test_OneLayer.py

```diff
@@ -3,104 +3,51 @@
 labeled data (first argument).  Results are saved to the path specified by the
 third argument.
 '''
+import argparse
 import json
 import logging
 import os
-import sys
 import time
 
-import numpy as np
+import poseidonml.training_utils as utils
 from poseidonml.Model import Model
 
 
 logging.basicConfig(level=logging.INFO)
-
-
-def calc_f1(results, logger, ignore_unknown=False):
-    results_by_label = {}
-    for file, file_results in results.items():
-        if file != 'labels':
-            indiv_results = file_results['individual']
-            true_label = file_results['label']
-
-            if true_label not in results_by_label:
-                if true_label == 'Unknown':
-                    if ignore_unknown is False:
-                        results_by_label[true_label] = {
-                            'tp': 0, 'fp': 0, 'fn': 0}
-                else:
-                    results_by_label[true_label] = {'tp': 0, 'fp': 0, 'fn': 0}
-
-            for _, classification in indiv_results.items():
-                class_label = classification[0][0]
-                if class_label == 'Unknown' and ignore_unknown is True:
-                    class_label = classification[1][0]
-                if class_label not in results_by_label:
-                    results_by_label[class_label] = {'tp': 0, 'fp': 0, 'fn': 0}
-                if true_label != 'Unknown':
-                    if class_label == true_label:
-                        results_by_label[true_label]['tp'] += 1
-                    if class_label != true_label:
-                        results_by_label[true_label]['fn'] += 1
-                        results_by_label[class_label]['fp'] += 1
-                elif ignore_unknown is False:
-                    if class_label == true_label:
-                        results_by_label[true_label]['tp'] += 1
-                    if class_label != true_label:
-                        results_by_label[true_label]['fn'] += 1
-                        results_by_label[class_label]['fp'] += 1
-    f1s = []
-    for label in results_by_label:
-        tp = results_by_label[label]['tp']
-        fp = results_by_label[label]['fp']
-        fn = results_by_label[label]['fn']
-
+logger = logging.getLogger(__file__.split(os.path.sep)[-1])
+
+
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--config', '-c', default='opts/config.json',
+                        help='model\'s config file')
+    parser.add_argument('--pcaps', '-P', default='/pcaps',
+                        help='pcap directory to train on (e.g., /pcaps)')
+    parser.add_argument('--model', '-r', default='/models/OneLayerModel.pkl',
+                        help='path to the RandomForest model (pickle file)')
+    parser.add_argument('--save', '-w', default='models/OneLayerModel.json',
```

Comment:
is this a thing yet?

Reply:
The json file is a functionality that's still being worked on. The pkl file generated by training is still in play, but I'd like to get that changed over to json sometime soon.